### PR TITLE
make _time_remaining a property

### PR DESCRIPTION
**Pull request by mambocab**
This addresses PYTHON-755, in which `_time_remaining` is updated incorrectly, by avoiding updating it at all.

### CODE REVIEW

File: cassandra/cluster.py [hunk 1]
```diff
@@ -3334,6 +3333,12 @@ def __init__(self, session, message, query, timeout, metrics=None, prepared_stat
         self.attempted_hosts = []
         self._start_timer()
 
+    @property
+    def _time_remaining(self):
+        if self.timeout is None:
+            return None
```
Comment (aboudreault) [inline cassandra/cluster.py]: Should you return 0 here?

File: cassandra/cluster.py [hunk 2]
```diff
@@ -3364,8 +3369,6 @@ def _on_speculative_execute(self):
         self._timer = None
         if not self._event.is_set():
             if self._time_remaining is not None:
```
Comment (aboudreault) [inline cassandra/cluster.py]: I see.... you could remove this IF if you just return 0 in _time_remaining I think.

**Comment by aboudreault** [comment]
minor comments, +1

**Comment by mambocab** [comment]
`timeout`, and by extension `_time_remaining`, can be `None` to signify "never time out" -- see [this timeout-handling code](https://github.com/datastax/python-driver/blob/master/cassandra/cluster.py#L3391).

There are some test failures -- some are transient Jenkins issues relating to pulling from the ASF's git server, and the other is also flaky on master.

Thanks for the review, merging.

**Pull request merged**